### ISSUE TITLE
proxyReq.removeHeader throws an error on tab load

### ISSUE THREAD

**Issue by zjpersc** [immobiliare/backstage-plugin-gitlab#143]
Running Backstage v1.12.0 and v5.0.0 of the plugin. 

When loading the tab associated with the plugin's content, the following error is thrown:

```
 node:_http_outgoing:771
[1]     throw new ERR_HTTP_HEADERS_SENT('remove');
[1]     ^
[1]
[1] Error [ERR_HTTP_HEADERS_SENT]: Cannot remove headers after they are sent to the client
[1]     at new NodeError (node:internal/errors:400:5)
[1]     at ClientRequest.removeHeader (node:_http_outgoing:771:11)
[1]     at ProxyServer.onProxyReq (C:\dev\van-1-12-0\node_modules\@immobiliarelabs\backstage-plugin-gitlab-backend\dist\index.cjs.js:36:20)
[1]     at ProxyServer.emit (C:\dev\van-1-12-0\node_modules\eventemitter3\index.js:184:35)
[1]     at ClientRequest.<anonymous> (C:\dev\van-1-12-0\node_modules\http-proxy\lib\http-proxy\passes\web-incoming.js:133:16)
[1]     at ClientRequest.emit (node:events:513:28)
[1]     at ClientRequest.emit (node:domain:489:12)
[1]     at tickOnSocket (node:_http_client:834:7)
[1]     at onSocketNT (node:_http_client:897:5)
[1]     at process.processTicksAndRejections (node:internal/process/task_queues:83:21) {
[1]   code: 'ERR_HTTP_HEADERS_SENT'
[1] }
[1]
[1] Node.js v18.13.0
```

I have been able to mitigate the error by commenting out the following code:
```
//@immobilarielabs/backstage-plugin-gitlab-backend/dist/index.cjs/js (L35-37)
        onProxyReq: (proxyReq) => {
          proxyReq.removeHeader("Authorization");
        }
```

I assume the code is necessary, so I don't view that as likely a long term fix.

**Comment by antoniomuso** [comment]
Hi @zjpersc, I searched for the error and found this issue related to our proxying plugin:https://github.com/chimurai/http-proxy-middleware/issues/343 https://github.com/http-party/node-http-proxy/issues/1287 

I have to find a way to reproduce locally the same error.

**Comment by zjpersc** [comment]
@antoniomuso I will say that I validated adding the 'proxyTimeout: 20000' to the httpProxyMiddleware options works for me. If you would like I can submit that as a PR to this project, but it is just one line here:

https://github.com/immobiliare/backstage-plugin-gitlab/blob/main/packages/gitlab-backend/src/service/router.ts

After Line 35:
`changeOrigin: true,`

Add:
`proxyTimeout: 20000,`

**Comment by zjpersc** [comment]
@antoniomuso Catching the error w/o doing the proxyTimeout works as well. 

My code:
```
        onProxyReq: (proxyReq) => {
          try {
            proxyReq.removeHeader("Authorization");
          } catch (e) {
            console.log((e instanceof Error).message);
          }
```

**Comment by antoniomuso** [comment]
Hi, catching seems to me a good solution, we can also add a custom timeout as optional parameter in plugin configuration.

**Comment by fmenesesg** [comment]
The catch didn't fix the problem, I just get 401 on every request, I think that something similar to what backstage did with the problem should be implemented: https://github.com/backstage/backstage/blob/f1c2ff2cdb01fa8c7d02e4d224ae5513e4a6b542/plugins/proxy-backend/src/service/router.ts#L144

**Comment by antoniomuso** [comment]
LGTM! Good catch @fmenesesg

**Comment by antoniomuso** [comment]
closed by #184

**Comment by mbenson** [comment]
Please see my comment https://github.com/immobiliare/backstage-plugin-gitlab/commit/9a302957d78664bc5146ce1334f54d0e7ed0e908#commitcomment-117894397
